src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -50,9 +50,11 @@ internal class ConnectionManager : IConnectionManager {
 
             _disposableBag = DisposableBag.Create<ConnectionManager>()
                 .Add(_statusBarViewModel)
-                .Add(() => _sessionProvider.BrokerStateChanged -= BrokerStateChanged);
+                .Add(() => interactiveWorkflow.RSession.Connected -= RSession_Connected)
+                .Add(() => interactiveWorkflow.RSession.Disconnected -= RSession_Disconnected);
 
-            _sessionProvider.BrokerStateChanged += BrokerStateChanged;
+            interactiveWorkflow.RSession.Connected += RSession_Connected;
```

Comment:
This is wrong - user will get notification that connection is broken every time RHost for Interactive Window RSession dies.